transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/scheduling/persistence/SchemaManager.java

```diff
@@ -47,34 +50,28 @@
           + "FOREIGN KEY (context_id) REFERENCES transport_contexts(_id) ON DELETE CASCADE)";
 
   private static final String CREATE_EVENT_METADATA_SQL_V1 =
-      "CREATE TABLE event_metadata "
+      "CREATE TABLE IF NOT EXISTS event_metadata "
           + "(_id INTEGER PRIMARY KEY,"
           + " event_id INTEGER NOT NULL,"
           + " name TEXT NOT NULL,"
           + " value TEXT NOT NULL,"
           + "FOREIGN KEY (event_id) REFERENCES events(_id) ON DELETE CASCADE)";
 
   private static final String CREATE_CONTEXTS_SQL_V1 =
-      "CREATE TABLE transport_contexts "
+      "CREATE TABLE IF NOT EXISTS transport_contexts "
           + "(_id INTEGER PRIMARY KEY,"
           + " backend_name TEXT NOT NULL,"
           + " priority INTEGER NOT NULL,"
           + " next_request_ms INTEGER NOT NULL)";
 
   private static final String CREATE_EVENT_BACKEND_INDEX_V1 =
-      "CREATE INDEX events_backend_id on events(context_id)";
+      "CREATE INDEX IF NOT EXISTS events_backend_id on events(context_id)";
 
   private static final String CREATE_CONTEXT_BACKEND_PRIORITY_INDEX_V1 =
-      "CREATE UNIQUE INDEX contexts_backend_priority on transport_contexts(backend_name, priority)";
-
-  private static final String DROP_EVENTS_SQL = "DROP TABLE events";
-
-  private static final String DROP_EVENT_METADATA_SQL = "DROP TABLE event_metadata";
-
-  private static final String DROP_CONTEXTS_SQL = "DROP TABLE transport_contexts";
+      "CREATE UNIQUE INDEX IF NOT EXISTS contexts_backend_priority on transport_contexts(backend_name, priority)";
 
   private static final String CREATE_PAYLOADS_TABLE_V4 =
-      "CREATE TABLE event_payloads "
+      "CREATE TABLE IF NOT EXISTS event_payloads "
```

Comment:
One of the goals was to have the system be lossless.
Are there a specific sequence of events that you can describe that would lead to faulty schemas?

I realize this is a mental exercise, but that's the only tool we have to reason about this.
Here's the PR that demonstrates the failures that we are currently prone to : https://github.com/firebase/firebase-android-sdk/pull/1548